addons/binding/org.openhab.binding.magentatv/src/main/java/org/openhab/binding/magentatv/internal/MagentaTVControl.java

```diff
@@ -0,0 +1,585 @@
+     *
+     * @param runStatus Integer code parsed form json (see EV_EITCHG_RUNNING_XXX)
+     * @return runningStatus as String
+     */
+    public String getRunStatus(int runStatus) {
+        if (runStatus == EV_EITCHG_RUNNING_NOT_RUNNING) {
+            return "stopped";
+        } else if (runStatus == EV_EITCHG_RUNNING_STARTING) {
+            return "starting";
+        } else if (runStatus == EV_EITCHG_RUNNING_PAUSING) {
+            return "paused";
+        } else if (runStatus == EV_EITCHG_RUNNING_RUNNING) {
+            return "running";
+        }
+        return Integer.toString(runStatus);
+    }
+
+    /**
+     * builds url from the discovered IP address/port and the requested uri
+     *
+     * @param uri requested URI
+     * @return the complete URL
+     */
+    public String buildReceiverUrl(String uri) {
+        return MessageFormat.format("http://{0}:{1}{2}", thingConfig.getIpAddress(), thingConfig.getPort(), uri);
+    }
+
+    /**
+     * build host string
+     *
+     * @return formatted string (<ip_address>:<port>)
+     */
+    private String buildHost() {
+        return thingConfig.getIpAddress() + ":" + thingConfig.getPort();
+    }
+
+    /**
+     * Given a string, return the MD5 hash of the String.
+     *
+     * @param unhashed The string contents to be hashed.
+     * @return MD5 Hashed value of the String. Null if there is a problem hashing
+     *         the String.
+     */
+    public static String computeMD5(String unhashed) {
+        try {
+            byte[] bytesOfMessage = unhashed.getBytes(CHARSET_UTF8);
+
+            MessageDigest md5 = MessageDigest.getInstance(HASH_ALGORITHM_MD5);
+            byte[] hash = md5.digest(bytesOfMessage);
+            StringBuilder sb = new StringBuilder(2 * hash.length);
+            for (byte b : hash) {
+                sb.append(String.format("%02x", b & 0xff));
+            }
+
+            return sb.toString();
+        } catch (Exception exp) {
+            return null;
+        }
+    }
+
+    /**
+     * Helper to parse a Xml tag value from string without using a complex XML class
+     *
+     * @param xml     Input string in the format <tag>value</tag>
+     * @param tagName The tag to find
+     * @return Tag value (between <tag> and </tag>)
+     */
+    public static String getXmlValue(String xml, String tagName) {
+        String open = "<" + tagName + ">";
+        String close = "</" + tagName + ">";
+        if (xml.contains(open) && xml.contains(close)) {
+            return StringUtils.substringBetween(xml, open, close);
+        }
+        return "";
+    }
+
+    /**
+     * Initialize key map (key name -> key code)
+     *
+     * for a list of valid key codes see
+     * http://support.huawei.com/hedex/pages/DOC1100366313CEH0713H/01/DOC1100366313CEH0713H/01/resources/dsv_hdx_idp/DSV/en/en-us_topic_0094619112.html
+     */
+    private void initializeKeyMap() {
```

Comment:
I suggest to move those static values into a resource file. Easier to modify and extend in the future.
You can access a resource file via FrameworkUtils and map it to a `Map` in just one line of code.